logsmanagement/logsmanagement.c

```diff
@@ -0,0 +1,637 @@
+             * try log format autodetection based on last log file line.
+             * TODO 1: Add another case in OR where log_format is compared with a valid reg exp.
+             * TODO 2: Set default log format and delimiter if not found in config? Or auto-detect? */ 
+            if(!log_format || !strcmp(log_format, "auto")){ 
+                info("Attempting auto-detection of log format for:%s", p_file_info->filename);
+
+                char *line = read_last_line(p_file_info->filename, 0);
+                if(!line){
+                    freez(p_file_info->parser_config);
+                    goto next_section; // TODO: Terminate monitor_log_file_init() if !parser_buff->line? 
+                }
+
+                p_file_info->parser_config->gen_config = auto_detect_web_log_parser_config(line, delimiter);
+                freez(line);
+            }
+            else{
+                p_file_info->parser_config->gen_config = read_web_log_parser_config(log_format, delimiter);
+                freez(log_format);
+                info("Read web log parser config for %s: %s\n", p_file_info->filename, 
+                                                                p_file_info->parser_config ? "success!" : "failed!");
+            }
+
+            if(!p_file_info->parser_config->gen_config){
+                // TODO: Terminate monitor_log_file_init() if p_file_info->parser_config is NULL? 
+                freez(p_file_info->parser_config);
+                goto next_section; 
+            } 
+
+            /* Check whether metrics verification during parsing is required */
+            Web_log_parser_config_t *wblp_config = (Web_log_parser_config_t *) p_file_info->parser_config->gen_config;
+            wblp_config->verify_parsed_logs = appconfig_get_boolean( &log_management_config, 
+                                                                     config_section->name, 
+                                                                     "verify parsed logs", 0);
+            info( "Log parsing verification: %s (%d) for %s.", 
+                  wblp_config->verify_parsed_logs ? "enabled" : "disabled", 
+                  wblp_config->verify_parsed_logs, config_section->name);
+            
+            for(int j = 0; j < wblp_config->num_fields; j++){
+                if((wblp_config->fields[j] == VHOST_WITH_PORT || wblp_config->fields[j] == VHOST) 
+                    && appconfig_get_boolean(&log_management_config, config_section->name, "vhosts chart", 0)){ 
+                    p_file_info->parser_config->chart_config |= CHART_VHOST;
+                }
+                if((wblp_config->fields[j] == VHOST_WITH_PORT || wblp_config->fields[j] == PORT) 
+                    && appconfig_get_boolean(&log_management_config, config_section->name, "ports chart", 0)){ 
+                    p_file_info->parser_config->chart_config |= CHART_PORT;
+                }
+                if((wblp_config->fields[j] == REQ_CLIENT) 
```

Comment:
Considering that three consecutive `ifs` have `REQ_CLIENT` as one of conditions, I suggest to test this only one time, and move the other conditions inside a `if((wblp_config->fields[j] == REQ_CLIENT)`